src/v0/util/index.js

```diff
@@ -2136,6 +2136,96 @@ const parseConfigArray = (arr, key) => {
   return arr.map((item) => item[key]);
 };
 
+/**
+ * Finds an existing batch based on metadata JobIds from the provided batch and metadataMap.
+ * @param {*} batch
+ * @param {*} metadataMap The map containing metadata items indexed by JobIds.
+ * @returns
+ */
+const findExistingBatch = (batch, metadataMap) => {
+  let existingBatch = null;
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const metadataItem of batch.metadata) {
+    if (metadataMap.has(metadataItem.jobId)) {
+      existingBatch = metadataMap.get(metadataItem.jobId);
+      break;
+    }
+  }
+
+  return existingBatch;
+};
+
+/**
+ * Removes duplicate metadata within each merged batch object.
+ * @param {*} mergedBatches An array of merged batch objects.
+ */
+const removeDuplicateMetadata = (mergedBatches) => {
+  mergedBatches.forEach((batch) => {
+    const metadataSet = new Set();
+    // eslint-disable-next-line no-param-reassign
+    batch.metadata = batch.metadata.filter((metadataItem) => {
+      if (!metadataSet.has(metadataItem.jobId)) {
+        metadataSet.add(metadataItem.jobId);
+        return true;
+      }
+      return false;
+    });
+  });
+};
+
+/**
+ * Combines batched requests with the same JobIds.
+ * @param {*} inputBatches The array of batched request objects.
+ * @returns  The combined batched requests with merged JobIds.
+ *
+ */
+const combineBatchRequestsWithSameJobIds = (inputBatches) => {
+  const combineBatches = (batches) => {
+    const clonedBatches = [...batches];
+    const mergedBatches = [];
+    const metadataMap = new Map();
+
+    clonedBatches.forEach((batch) => {
+      const existingBatch = findExistingBatch(batch, metadataMap);
+
+      if (existingBatch) {
+        // Merge batchedRequests arrays
+        existingBatch.batchedRequest = [
+          ...(Array.isArray(existingBatch.batchedRequest)
```

Comment:
https://github.com/rudderlabs/rudder-transformer/blob/b5d28a315e859f159dc4b58e547393e19da592db/src/util/common.js

there is function toArray, use that to simplify this code